note.h

```diff
@@ -136,17 +136,32 @@ void NoteSetFnNoteMutex(mutexFn lockNotefn, mutexFn unlockNotefn);
 void NoteSetFnDefault(mallocFn mallocfn, freeFn freefn, delayMsFn delayfn, getMsFn millisfn);
 void NoteSetFn(mallocFn mallocfn, freeFn freefn, delayMsFn delayfn, getMsFn millisfn);
 void NoteSetFnSerial(serialResetFn resetfn, serialTransmitFn writefn, serialAvailableFn availfn, serialReceiveFn readfn);
+void NoteSetFnI2C(uint32_t i2caddr, uint32_t i2cmax, i2cResetFn resetfn, i2cTransmitFn transmitfn, i2cReceiveFn receivefn);
+void NoteSetFnDisabled(void);
+void NoteSetI2CAddress(uint32_t i2caddress);
 
+// The Notecard, whose default I2C address is below, uses a serial-to-i2c protocol whose "byte count"
+// must fit into a single byte and which must not include a 2-byte header field.  This is why the
+// maximum that can be transmitted by note-c in a single I2C I/O is 255 - 2 = 253 bytes.
 #define NOTE_I2C_ADDR_DEFAULT	0x17
-#ifndef NOTE_I2C_MAX_DEFAULT
-#define NOTE_I2C_MAX_DEFAULT	30
-#endif
 #ifndef NOTE_I2C_MAX_MAX
-#define NOTE_I2C_MAX_MAX		127
+#define NOTE_I2C_MAX_MAX		253
```

Comment:
Can we break `NOTE_I2C_MAX_MAX` into two parts?

```c
#define NOTE_I2C_HEADER_SIZE 2
#define NOTE_I2C_MAX_MAX (UCHAR_MAX - NOTE_I2C_HEADER_SIZE)
```

Reply:
Yea that's fine go for it